### PR TITLE
Limit the upsell to dotcom users & switch to ff 

### PR DESCRIPTION
**Pull request by mkondratek**
## Test plan
- play with testing accounts (with rate limit exceeded)
- remember to restart your ide (in particular kill & runIde again) when switching accounts

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/chat/Chat.kt [hunk 1]
```diff
@@ -77,30 +79,29 @@ class Chat {
...
+                      "Upgrade to Cody Pro for unlimited autocompletes, chats, and commands. " +
+                      "<a href=\"https://sourcegraph.com/cody/subscription\">Upgrade</a> " +
+                      "or <a href=\"https://sourcegraph.com/cody/subscription\">learn more</a>.<br><br>" +
+                      "(Already upgraded to Pro? Restart your IDE for changes to take effect)"
+                }
+                else -> {
+                  "<b>Request failed:</b> You've used all${rateLimitError.quotaString()} chat messages and commands." +
+                      " The allowed number of request per day is limited at the moment to ensure the service stays functional.${rateLimitError.resetString()} " +
+                      "<a href=\"https://docs.sourcegraph.com/cody/core-concepts/cody-gateway#rate-limits-and-quotas\">Learn more.</a>"
+                }
```
Comment (exigow) [inline src/main/kotlin/com/sourcegraph/cody/chat/Chat.kt:98]: We can use a multiline string everywhere for readability.

**Comment by Sa1to** [comment]
I don't see any check if user is dotcom...

File: src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt [hunk 2]
```diff
@@ -57,25 +63,36 @@ private constructor(content: String, shouldShowUpgradeOption: Boolean) :
...
-          }
-      return UpgradeToCodyProNotification(content, shouldShowUpgradeOption)
+        UpgradeToCodyProNotification(content, shouldShowUpgradeOption).notify(project)
+      }
+    }
+
+    @RequiresEdt
+    fun isCodyProJetbrains(project: Project): Boolean {
+      val activeAccountType = CodyAuthenticationManager.instance.getActiveAccount(project)
+      if (activeAccountType != null && activeAccountType.isDotcomAccount()) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt:88]: @Sa1to the check is here 🕵️

**Review by Sa1to — APPROVED** [review]
LGTM

**Review by exigow — APPROVED** [review]
Looks good :+1: , but I'm not sure that the changes with the feature flag have been working correctly.

**Pull request merged**